[FM v0.8.0]--Add builtin connectors docs (#204)
streamnative/function-mesh-website: docs/function-mesh-worker/manage-builtin-connectors.md

Pulsar distribution includes a set of common connectors that have been packaged and tested with the rest of Apache Pulsar. These connectors import and export data from some of the most commonly used data systems.

Using any of these connectors is as easy as writing a simple connector and running the connector locally or submitting the connector to a Pulsar Functions cluster.

Pulsar Functions Worker provides `sinks` and `sources` APIs to let the user manage Pulsar IO Connectors, it also provides a built-in connectors feature, with this feature, user do not need to provide the connectors' NAR package file and uses the connector type directly. From the Pulsar Functions Worker side, it will load the packages from `connectorsDirectory` (defaults to `./connectors`) and uses the connectors as built-in connectors list.

## Customize the built-in connectors in Function Mesh Worker Service

With Function Mesh Worker service, it requires a customized YAML file placed into `conf/connectors.yaml` with content of the metadata of each built-in connector. Below are the configureable fields of built-in connector:

| Field | Description |
| ---|---|
| `id` | The id of the IO connector. |

> | `name` | The name of the connector type. |

Reviewer: Do you mean that the name of the connector can be set to source or sink?

> | `description` | Description to be used for user help. |

Reviewer: ```suggestion
| `description` | The description that is used for user help. |
```

> | `sourceClass` | The class name for the connector source implementation. If not defined, it will be assumed this connector cannot act as a data source. |

Reviewer: ```suggestion
| `sourceClass` | The class name for the connector source implementation. If not defined, it assumes that the connector cannot act as a data source. |
```

> | `sinkClass` | The class name for the connector sink implementation. If not defined, it will be assumed this connector cannot act as a data sink. |

Reviewer: ```suggestion
| `sinkClass` | The class name for the connector sink implementation. If not defined, it assumes that the connector cannot act as a data sink. |
```

> | `sourceConfigClass` | The class name for the source config implementation. If not defined, the framework will not be able to do any submission time checks. |

Reviewer: ```suggestion
| `sourceConfigClass` | The class name for the source configuration implementation. If not defined, the framework cannot do any submission time checks. |
```

> | `sinkConfigClass` | The class name for the sink config implementation. If not defined, the framework will not be able to do any submission time checks. |

Reviewer: ```suggestion
| `sinkConfigClass` | The class name for the sink configuration implementation. If not defined, the framework cannot do any submission time checks. |
```

| `version` | The version of the connector. |

> | `imageRegistry` | The imageRegistry where host the connector image. By default the imageRegistry is empty, which refer to Docker Hub `docker.io/`. |

Reviewer: ```suggestion
| `imageRegistry` | The image registry that hosts the connector image. By default, the image registry is empty, which means that it uses the Docker Hub (`docker.io/`) to host the connector image. |
```

> | `imageRepository` | The imageRepository to the connector. Usually it in format of NAMESPACE/REPOSITORY. |

Reviewer: ```suggestion
| `imageRepository` | The image repository to the connector. Usually, it is in a format of NAMESPACE/REPOSITORY. |
```

> | `imageTag` | The imageTag to the connector image. By default it will align with Pulsar's version. |

Reviewer: ```suggestion
| `imageTag` | The tag to the connector image. By default, it aligns with Pulsar's version. |
```

> | `typeClassName` | Type class name of the connector or function. If not set, the default value '[B' will be used |

Reviewer: ```suggestion
| `typeClassName` | The type class name of the connector or function. By default, it is set to `'[B'`.|
```

> | `sourceTypeClassName` | Type class name of the source connector. If not set, will inherit the value from typeClassName. |

Reviewer: ```suggestion
| `sourceTypeClassName` | The type class name of the source connector. If not set, it will inherit the value from the `typeClassName` field. |
```

> | `sinkTypeClassName` | Type class name of the sink connector. If not set, will inherit the value from typeClassName. |

Reviewer: ```suggestion
| `sinkTypeClassName` | The type class name of the sink connector. If not set, it will inherit the value from the `typeClassName` field. |
```

> | `jarFullName` | JAR or NAR package name of the connector, optional. If not set, will be generated by id and version. | 

Reviewer: ```suggestion
| `jarFullName` | Optional. The JAR or NAR package name of the connector. If not set, it is generated based on the connector ID and connector version. | 
```

> | `defaultSchemaType` | Default schema type of the connector's topic, optional. |

Reviewer: ```suggestion
| `defaultSchemaType` | Optional. The default schema type of the connector's topic. |
```

> | `defaultSerdeClassName` | Default serde class name of the connector's topic, optional. |

Reviewer: ```suggestion
| `defaultSerdeClassName` | Optional. The default serde class name of the connector's topic. |
```

## Sample YAML file

> 

Reviewer: This sample YAML file lists configurations of all currently-supported built-in connectors.

Author: thanks, and addressed with the latest commit, PTAL

> ```

Reviewer: ```suggestion
```yaml
```

---
title: Manage built-in connectors
category: function-mesh-worker
id: manage-builtin-connectors
---

This document describes how to manage built-in Connectors with Function Mesh Worker service. 

> ## Built-in Connector feature in Apache Pulsar

Reviewer: ```suggestion
## Built-in connectors in Apache Pulsar
```
